atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ReadTransaction.java

```diff
@@ -149,11 +150,13 @@ public Iterator<Map.Entry<Cell, byte[]>> getRowsColumnRange(
 
     private void checkTableName(TableReference tableRef) {
         Optional<SweeperStrategy> sweepStrategy = sweepStrategies.get(tableRef).getSweeperStrategy();
-        Preconditions.checkState(
-                sweepStrategy.isEmpty() || sweepStrategy.get().equals(SweeperStrategy.CONSERVATIVE),
-                "This table cannot be read from a read-only transaction, because its sweep strategy is neither NOTHING"
-                        + " nor CONSERVATIVE",
-                LoggingArgs.tableRef(tableRef));
```

Comment:
this was secondary in my analysis

Reply:
yeah for this one it's very dependent on read transaction usage (but fires on every get in a read transaction)